### PR TITLE
WIP: ubuntu 14.04 ppc64le

### PR DESCRIPTION
**Pull request by jBarz**
Enable ubuntu 14.04 ppc64le.
Have both gcc 4.8 and 4.9 on machine available

### CODE REVIEW

File: ansible/roles/java-base/vars/main.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ packages: {
   'freebsd': 'openjdk8-jre',
   'smartos': 'openjdk8',
   'ubuntu': 'openjdk-8-jre-headless',
-  'ubuntu1404': 'oracle-java8-installer',
+  'ubuntu1404': 'openjdk-7-jre-headless',
```
Comment (mhdawson) [inline ansible/roles/java-base/vars/main.yml]: We need Java 8 on all of the jenkins nodes, regardless of the OS level as that is the minimum that works with our levels of Jenkins.

We had to manually update the PPC machines a while back, we should use that template to update these ansible scripts to get the right version on the PPC machines as well.

The templates are here: https://github.com/nodejs/build/issues/775#issuecomment-336245389

Comment (gdams) [reply ansible/roles/java-base/vars/main.yml]: If we wish to fetch the ppcle binaries from AdoptOpenJDK then we should consider using the api https://api.adoptopenjdk.net/README

Comment (jBarz) [reply ansible/roles/java-base/vars/main.yml]: Fixed

**Comment by mhdawson** [comment]
If you have used this to configure ubuntu1404-ppc64_le-3 I think the next step is to connect it up to the so that we can validate the machine works as a test worker.

The one thing we should probably test as part of that is the selection of the 4.9 versus 4.8 compiler.  We should either have a script in the build repo for that or have that script isntalled as part of the ansible setup.  It may make sense to be part of what is installed by the ansible scirpts as we'd need to update when we add support for a new compiler level which would also require ansible script updates.

**Comment by jBarz** [comment]
I have run the ansible script successfuly on ubuntu1404-ppc64_le-3.
To build node using gcc-v4.8, we need to set the following. Equivalent for 4.9.

export CC=/usr/bin/gcc-4.8
export CXX=/usr/bin/g++-4.8
export LINK=/usr/bin/g++-4.8
export LDFLAGS=“-Wl,-rpath,$(dirname `$CC --print-file-name libgcc_s.so`)”

**Comment by mhdawson** [comment]
I think next step is create test CI job as clone of existing job and do run against ubuntu1404-ppc64_le-3.

I would like to check in a script into the build repo called something like set_gcc_level that would looks at the files checked out for Node.js (likely the version info) and then sets the required exports.

The main downside is that we'd need a clone of the build repo but I don't think that would be a big deal.  @nodejs/build any concerns?

**Comment by maclover7** [comment]
Checking scripts into the build repo is fine by me. https://github.com/nodejs/build/tree/master/jenkins/scripts has some prior work, if helpful at all.

**Comment by maclover7** [comment]
ping @jBarz

**Comment by mhdawson** [comment]
@jBarz  The paths don't seem to be quite right based on the ansible setup.  I had to add the path to java 8 to get the jenkins agent to start: /home/iojs/jdk8u144-b01/jre/bin/java in jenkins.conf.

jenkins.conf probably either needs to have the path updated like I did or probably better the path so that it include/home/iojs/jdk8u144-b01/jre/bins

**Comment by mhdawson** [comment]
Testing builds on le-3 here: 
https://ci.nodejs.org/job/node-test-commit-plinux_mdawson/21/
https://ci.nodejs.org/job/node-test-commit-plinux_mdawson/22/
https://ci.nodejs.org/job/node-test-commit-plinux_mdawson/23/

**Comment by mhdawson** [comment]
Tests ran/passed for the 3 combinations.

**Comment by mhdawson** [comment]
PR for script to select compiler level: https://github.com/nodejs/build/pull/1103 that was tested in runs above on be-3.

**Comment by jBarz** [comment]
I modified the ansible script to append "/home/iojs/.profile" with
`export PATH=/home/iojs/jdk8u144-b01:\$PATH`

Is this an acceptable solution?
We are assuming that Jenkins will only run as "iojs".

**Comment by maclover7** [comment]
@jBarz Seems reasonable to me.

@mhdawson Can you run these scripts again on all ppc machines and confirm everything's ok? After that, please land at will.

**Comment by mhdawson** [comment]
@maclover7 can you review/approve https://github.com/nodejs/build/pull/1103 as its related to this work.

**Comment by maclover7** [comment]
@mhdawson Approved 1103. Once this is ready, please feel free to land in master, and then please run on all PPC machines to confirm everything's OK.

**Comment by mhdawson** [comment]
Unfortunately the change to set the path does not quite work as the script that starts the worker sets the path directly. Will discuss with @jBarz to figure out how to fix.

**Comment by mhdawson** [comment]
Looks good with latest update from @jBarz

**Comment by mhdawson** [comment]
Landed as 443f72ec12b3b9c05b4f7a74972680e4a5358ca5

**Pull request closed without merging**